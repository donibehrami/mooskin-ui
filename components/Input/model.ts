import { IDivBoxComponentProps, IInputBoxComponentProps } from '../Box/model';
import { IInputCallbackData } from '../index/index';

export interface IInputContainerComponentProps extends IDivBoxComponentProps {
    /** provide to make the input field disabled */
    disabled?: boolean;

    /** value to be passed down to the input */
    value: any;

    /** what data is being used, helps whn extracting user input, you know on what field changes are made */
    dataLabel?: string;

    /** callback that is called when the input changes */
    onChangeInput?: (e: React.ChangeEvent<HTMLInputElement>, data: IInputCallbackData) => void;
}

export interface IInputComponentProps extends IInputBoxComponentProps {
    /** provide to make the input field disabled */
    disabled?: boolean;

    /** what data is being used, helps whn extracting user input, you know on what field changes are made */
    dataLabel?: string;

    /** min and max values for number values */
    minmax?: number[];

    /** numberType */
    numberType?: 'integer' | 'float';

    /** whether its wrapped by its container */
    wrapped?: boolean;

    /** callback that is called when the input changes */
    onChangeInput?: (e: React.ChangeEvent<HTMLInputElement>, data: IInputCallbackData) => void;
}

export interface IInputIconComponentProps extends IDivBoxComponentProps {
    /** onclick event handler when clicking on the icon */
    onClickIcon?: (e: React.MouseEvent<HTMLDivElement>) => void;
}

export interface IInputOverlayComponentProps extends IDivBoxComponentProps {
    /** onclick event handler when clicking on the icon */
    onClickOverlay?: (e: React.MouseEvent<HTMLDivElement>) => void;
}

export interface IInputEmojiComponentProps extends IDivBoxComponentProps {
    /** onclick event handler when clicking on the emoji */
    onChangeEmoji?: (value: string) => void;
}

export interface IInputListComponentProps extends IDivBoxComponentProps {
    /** icon to be shown for toggling the dropdown */
    icon: string;

    /** onclick event handler when clicking on the component */
    onClickOption?: (e: React.MouseEvent<HTMLDivElement>, value: string) => void;
}

export interface IInputOptionComponentProps extends IDivBoxComponentProps {
    /** value of the clicked element */
    value: string;

    /** onclick event handler when clicking on the component */
    onClickOption?: (value: string) => void;
}
