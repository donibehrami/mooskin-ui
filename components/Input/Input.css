@import '../_utils/globals/variables.css';
@import '../_utils/globals/screens.css';
@import url('https://fonts.googleapis.com/icon?family=Material+Icons');

.inputContainer{
    display: flex;
    margin: 3px;
    box-sizing: border-box;
    align-items: baseline;
}

.innerContainer{
    display: flex;
    flex: 1 0 auto;
    align-items: baseline;
}

.inputDiv{
    display: flex;
    flex-direction: column;
    box-sizing: border-box;
    flex: 1 0 auto;
}

.column {
    flex-direction: column;
    align-items: stretch;
}

.input{
    display: flex;
    outline: none;
    font-family: Arial,sans-serif;
    border: 0;
    color: var(--inputFontColor);
    /* font-size: 16px; */
    font-size: 1rem; 
    width: 100%;
    transition: 0.2s;
    background: transparent;
}

.innerDiv{
    display: flex;
    background: #efefef;
    border-radius: 8px;
}

.reverse{
    flex-direction: row-reverse;
}

.noIcon{
    padding: 15px;
    border-radius: 8px;
}

.inputLeft{
    border-top-left-radius: 8px;
    border-bottom-left-radius: 8px;
    padding: 15px 5px 15px 15px;
}

.inputRight{
    border-top-right-radius: 8px;
    border-bottom-right-radius: 8px;
    padding: 15px 15px 15px 5px;
}

.disabledInput{
    cursor:not-allowed;
    color: #bfbfbf;
    background: #f6f6f6;
}

.input::placeholder {
    color: var(--inputPlaceholder);
}

.iconLeft{
    border-top-left-radius: 8px;
    border-bottom-left-radius: 8px;
    padding: 10px 0 10px 10px;
}

.iconRight{
    border-top-right-radius: 8px;
    border-bottom-right-radius: 8px;
    padding: 10px 10px 10px 0;
}

.icon{
    color: var(--inputPlaceholder);
    font-size: 25px;
    font-family: 'Material Icons';
    background: transparent;
}

.iconError{
    color: #ec3232;
}

.iconSuccess{
    color: #00cc44;
}

.input:focus {
    outline: none;
    color:#333333;
}

.inputLabel {
    color: #6b6b6b;
     /* font-size: 13px;  */
     font-size: .92857142857rem; 
    cursor: pointer;
    padding-top: 15px;
    flex-basis: 150px;
}

.topLabel {
    flex-basis: auto;
    padding-top: 0px;
    margin-left: 3px;
}

.description{
    display: block;
    color: var(--italicDescription);
    font-size: .78571428571rem;
    margin-left: 3px;
    margin-top: 5px;
}

.success{
    box-shadow: inset 0px 0px 0px 2px #00cc44;
}

.error{
    box-shadow: inset 0px 0px 0px 2px #ec3232;
}

.descError{
    color: #ec3232;
}

.descSuccess{
    color: #00cc44;
}

.copyButton {
    margin-left: 15px;
}

@media (--xSmall){
    .inputContainer{
        flex-direction: column;
        align-items: stretch;
    }

    .innerContainer {
        flex-direction: column;
        align-items: stretch;
    }

    .copyButton{
        margin: 5px 0 0 0;
    }
    
    .inputLabel{
        flex-basis: auto;
        padding-top: 0px;
        margin-left: 3px;
    }

}
