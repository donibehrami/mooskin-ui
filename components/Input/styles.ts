import styled from 'styled-components';

// Models
import { IBoxComponentProps } from '../Box/model';
import { IInputComponentProps, IInputContainerComponentProps, IInputListComponentProps, IInputOptionComponentProps } from './model';

// Components
import { Box } from '../Box/Box';

import variables from '../_utils/globals/variables';

export const StyledInputContainer = styled(Box)<IInputContainerComponentProps>`
	position: relative;
	color: ${(props) => props.palette?.fontColors.common || variables.fontColors.common};
	display: flex;
	flex: 1;
	align-items: center;
	opacity: ${(props) => (props.disabled ? 0.5 : 1)};
	cursor: ${(props) => (props.disabled ? 'not-allowed' : 'pointer')};
	border-radius: 3px;
	border: solid 1px ${(props) => props.palette?.borderColors.common || variables.borderColors.common};
	background-color: ${(props) => props.palette?.backgroundColors.common || variables.backgroundColors.common};
	padding: 7px 15px 7px;
`;

StyledInputContainer.displayName = 'StyledInputContainer';

export const StyledInputBase = styled(Box)<IInputComponentProps>`
	font-family: Hind;
	font-size: 14px;
	font-weight: normal;
	font-stretch: normal;
	font-style: normal;
	letter-spacing: normal;
	text-align: left;
	flex: 1;
`;

StyledInputBase.displayName = 'StyledInputBase';

export const StyledInputSolo = styled(StyledInputBase)<IInputComponentProps>`
	border-radius: 3px;
	width: 100%;
	color: ${(props) => props.palette?.fontColors.common || variables.fontColors.common};
	border: solid 1px ${(props) => props.palette?.borderColors.common || variables.borderColors.common};
	background-color: ${(props) => props.palette?.backgroundColors.common || variables.backgroundColors.common};
	padding: 7px 15px 7px;
`;

StyledInputSolo.displayName = 'StyledInputSolo';

export const StyledInputWrapped = styled(StyledInputBase)<IInputComponentProps>`
	color: inherit;
	border: none;
	background-color: transparent;
`;

StyledInputWrapped.displayName = 'StyledInputWrapped';

export const StyledInputOverlay = styled(Box)<IBoxComponentProps>`
	position: fixed;
	top: 0;
	left: 0;
	right: 0;
	bottom: 0;
	z-index: -1;
`;

StyledInputOverlay.displayName = 'StyledInputOverlay';

export const StyledInputIcon = styled(Box)<IBoxComponentProps>`
	font-family: 'Mooskin Icons';
	font-size: 20px;
	display: flex;
	align-items: center;
	color: inherit;
`;

StyledInputIcon.displayName = 'StyledInputIcon';

export const StyledInputOptionList = styled(Box)<Partial<IInputListComponentProps>>`
	position: absolute;
	right: 0;
	height: auto;
	background: #fff;
	border-radius: 3px;
	z-index: 4;
	box-shadow: 0 1px 5px rgba(0, 0, 0, 0.32);
`;

StyledInputOptionList.displayName = 'StyledInputOptionList';

export const StyledInputOptionListTitle = styled(Box)<IBoxComponentProps>`
	padding: 5px;
	color: #bebebe;
	border-bottom: 1px solid #f1f1f1;
	background-color: #f5f7f9;
	text-align: center;
	border-top-left-radius: 5px;
	border-top-right-radius: 5px;
`;

StyledInputOptionListTitle.displayName = 'StyledInputOptionListTitle';

export const StyledInputOption = styled(Box)<IInputOptionComponentProps>`
	padding: 3px 5px;
	:not(:last-child) {
		border-bottom: 1px solid #f1f1f1;
	}
`;

StyledInputOption.displayName = 'StyledInputOption';
