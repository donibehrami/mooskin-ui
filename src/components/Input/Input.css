.input{
    padding: 15px;
    font-family: Arial,sans-serif;
    border: 0;
    color: #333333;
    font-size: 14px;
    border-radius: 8px;
    background: #efefef;
    display: table-cell;
    margin: 3px;
}

.disabledInput{
    cursor:not-allowed;
}

.input::-webkit-input-placeholder {
color: #834E2E !important;
}
 
.input:-moz-placeholder { /* Firefox 18- */
color: #834E2E !important;  
}
 
.input::-moz-placeholder {  /* Firefox 19+ */
color: #834E2E !important;  
}
 
.input:-ms-input-placeholder {  
color: #834E2E !important;  
}

.input:focus {
    outline: none;
}