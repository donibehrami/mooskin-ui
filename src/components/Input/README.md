# MooSkin UI - Input Component

The MooSkin Input Component works similar to the normal HTML <input/> element, but with pre-defined styling and attributes.

___

### Usage

To start using the Input Component first you have to Import it

```
Import {Input} from 'mooskin';
```

And then you can simply start using it by typing

```
<Input whatever />
```

For easy use, the components are named similar to normal HTML components but with a capital first letter.

Like the <input/> element it will accept given attributes and render differently based on the given attributes

### Examples

Passing the placeholder attribute

```
<Input placeholder="username" />
```

or making it disabled with a value, in this case the proper disabled class will be loaded (ex. disabling cursor)

```
<Input value="This one is disabled" disabled />
```

or pass it a function for event handling

```
<Input onChange={yourFunc} />
```

or just give it a custom style

```
<Input style={yourStyle} />
```

## Supported attributes

* `type` - type of the input (eg. text)
* `name` - name of the input
* `value` - text value of input
* `placeholder` - backtext description of input
* `size` - size of the input field
* `minlength` - min number of characters allowed
* `maxlength` - max number of characters allowed
* `disabled` - input field should be disabled
* `required` - input field is required
* `style` - input field style

#### For more

___

[MooSkin-UI](https://github.com/moosend/mooskin-ui)