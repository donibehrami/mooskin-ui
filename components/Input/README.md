# MooSkin UI - Input Component

The MooSkin Input Component works similar to the normal HTML `<input/>` element, but with pre-defined styling and attributes.

___

### Usage

To start using the Input Component first you have to Import it

```
Import {Input} from 'mooskin';
```
or modular import
```
// JS
import Input from 'mooskin/lib/Input';

// CSS
import 'mooskin/lib/Input/style.css';
```

And then you can simply start using it by typing

```
<Input />
```

For easy use, the components are named similar to normal HTML components but with a capital first letter.

Like the `<input/>` element it will accept given attributes and render differently based on the given attributes

### Examples

Passing the placeholder attribute

```
<Input placeholder="username" />
```

or making it disabled with a value, in this case the proper disabled class will be loaded (ex. disabling cursor)

```
<Input value="This one is disabled" disabled />
```

or pass it a function for event handling

```
<Input onChange={yourFunc} />
```

or just give it a custom style

```
<Input style={yourStyle} />
```

### Callback

The Input Component Callback will always return the value of the Input on each change. This can be used with a function passed via the `onChange` prop. For example, if u want to console log the callback value, pass a function to the `onChange` prop.

```
const logValue = (e, data) => {     // data is the callback object, which consists of value and a dataLabel(not required)
    console.log(data.value);
};

<Input onChange={logValue} />
```
In this case on each input change, the value will be console logged.

This can be used in various situations and combinations, for an enhanced development experience.

### Validation

The Input Component can be validated using the `status` and `validate` props. The `status` prop will accept an 'error' or 'success' string to apply the appropriate classes to the input, or none for that matter. The `validate` prop is the validation function which accepts a arg of an object containing `value`, `dataLabel` and `required` props of the component as properties, and depending on the received value, you can use any kind of validation you like. The `description` prop can also be used in the process to inform the user about validation errors. The validation will fire the first time onBlur, and every other time onChange.

#### Example

```
this.state = {
    status: '',
    message: '',
    value: ''
}

<Input
    value={this.state.value}
    validate={this.validate}
    status={this.state.status}
    description={this.state.message}
    onChange={this.onChange}
/>

onChange(e, data) {
    this.setState({value: data.value})
}

validate(data){
    if (data.value){
        if (data.value.length < 5){
            this.setState({status: 'error', message: 'Input should have 5 or more characters'});
        } else {
            this.setState({status: '', message: ''})
        }
    } else {
        this.setState({status: 'error', message: 'This input field is required'});
    }
}
```

In this case, when the user focuses the input and blurs away immediately, an error class will get applied to the input and the user will get informed that thee input field is required. When the user starts typing, the error message will change, saying that the input should have 5 or more characters, when this is done, the error classes will get removed. Alternatevely, when the validation passes, we can pass a success status to apply the related class.

<div class="playground-doc">

## Supported attributes 

* `id` - id of the element
* `type` - type of the input (eg. text)
* `name` - name of the input
* `value` - text value of input
* `placeholder` - backtext description of input
* `minlength` - min number of characters allowed
* `maxlength` - max number of characters allowed
* `disabled` - input field should be disabled
* `required` - input field is required
* `description` - input description (small italic bottom)
* `customDropdown` - adds a custom dropdown like the dropdown for emoji or personalization tags. Should consist of the follow object, `icon` which will appear as dropdown icon, `label`, which will appear as label & `content` which should be a JSX element or an array of such. The prop acceps an array of such dropdown objects aswell, rendering multiple dropdowns.
* `labelWidth` - label width (default 150 px)
* `autofocus` - autofocus specified input on page load
* `autocomplete` - toggle autocomplete
* `className` - css class
* `dataLabel` - label what kind of data 
* `style` - input field style
* `icon` - icon inside input (material icon names)
* `iconPosition` - left or right (right is default)
* `iconClass` - class to be applied to the icon
* `iconStyle` - styles to be applied to the icon
* `onChange` - callback to be triggered on input change
* `validate` - validate function
* `minmax` - an array of one or two number values, if array has one value, that will be set as the maximum allowed number, if array has two values, the first will be set as the minimum value, the second will be set as the maximum value (only works with type="number" input)
* `emoji` - adds emoji icon to the input, which opens a dropdown for adding emojis
* `personalizationTags` - adds personalization tags based on passed prop, should be an array of object {label: string, value: string}
& `closeOnTagAdd` - closes personalization tags dropdown when a personalization tag is added
* `status` - to be used with validate, to apply classes based on error or success
* `extraHtmlAttr` - extra html attributes applied to the input

</div>

Allthough these attributes are supported, all of them are optional.

#### For more

___

[MooSkin-UI](https://github.com/moosend/mooskin-ui)