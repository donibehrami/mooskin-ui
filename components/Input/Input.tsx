import * as React from 'react';

import styles from './Input.css';

import {IInputCallbackData} from '../_utils/types/commonTypes';

export interface IProps {

    /** override input id */
    id?: string;

    /** provide to make the input field disabled */
    disabled?: boolean;

    /** provide to make the input field required */
    required?: boolean;

    /** override input type */
    type?: string;

    /** override input name */
    name?: string;

    /** override input value */
    value?: string;

    /** override input placeholder */
    placeholder?: string;

    /** override input minlength */
    minlength?: number;

    /** override input maxlength */
    maxlength?: number;

    /** input label */
    label?: string;

    /** input description (small italic bottom) */
    description?: string;

    /** spacing between label and input */
    spacing?: number;

    /** override input styles */
    style?: {[key: string]: string|number};

    /** override input class */
    className?: string;

    /** what data is being used, helps whn extracting user input, you know on what field changes are made */
    dataLabel?: string;

    /** callback that is called when the input changes */
    onChange?: (e: React.ChangeEvent<HTMLInputElement>, data: IInputCallbackData) => void;

}

export interface IInputState{
    value: string;
}

class Input extends React.Component<IProps, IInputState> {

    public static defaultProps = {
        className: '',
        style: {}
    };

    constructor(props: IProps){
        super(props);

        this.state = {
            value: this.props.value ? this.props.value : ''
        };
    }

    public render(){

        const {
            id,
            disabled,
            required,
            type,
            name,
            placeholder,
            minlength,
            maxlength,
            style,
            className,
            label,
            description,
        } = this.props;

        const disabledInput = disabled ? styles.disabledInput : '';
        const spacing = label ?
                        !this.props.spacing ?
                        {marginLeft: '20px'} :
                        {marginLeft: `${this.props.spacing}px`} :
                        {};
        const inputStyle = {...spacing, ...style};

        return (
            <div className={`input-component ${className}`}>
                <label className={styles.inputLabel}>
                    {label}
                    <input
                        onChange={this.onChange}
                        id={id}
                        type={type}
                        name={name}
                        value={this.state.value}
                        placeholder={placeholder}
                        minLength={minlength}
                        maxLength={maxlength}
                        required={required}
                        disabled={disabled}
                        className={`input ${styles.input} ${disabledInput}`}
                        style={inputStyle}
                    />
                    <i>{description}</i>
                </label>
            </div>
        );
    }

    private onChange = (e: React.ChangeEvent<HTMLInputElement>) => {
        this.setState({value: e.target.value});
        !this.props.disabled &&
        this.props.onChange &&
        this.props.onChange(e, {value: this.props.value, dataLabel: this.props.dataLabel});
    }

}

export default Input;
