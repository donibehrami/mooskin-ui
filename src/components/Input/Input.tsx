import * as React from 'react';

import styles from './Input.css';

export interface IProps {

    /** override input id */
    id?: string;

    /** provide to make the input field disabled */
    disabled?: boolean;

    /** provide to make the input field required */
    required?: boolean;

    /** override input type */
    type?: string;

    /** override input name */
    name?: string;

    /** override input value */
    value?: string;

    /** override input placeholder */
    placeholder?: string;

    /** override input size */
    size?: string;

    /** override input minlength */
    minlength?: string;

    /** override input maxlength */
    maxlength?: string;

    /** override input styles */
    style?: {[key: string]: string};

    /** override input class */
    className?: string;

    /** callback that is called when the input changes */
    onChange?: (e?: React.ChangeEvent<HTMLInputElement>) => void;

}

class Input extends React.Component<IProps, void> {

    public static defaultProps = {
        style: {},
        className: ''
    };

    public render(){

        const {id, disabled, required, type, name, value, placeholder, size, minlength, maxlength, style, className} = this.props;

        const disabledInput = disabled ? styles.disabledInput : '';
        let hasSize;
        let hasMin;
        let hasMax;

        if(size){
            hasSize = parseInt(size);
        }

        if(minlength){
            hasMin = parseInt(minlength);
        }

        if(maxlength){
            hasMax = parseInt(maxlength);
        }

        return (
            <input
                onChange={this.onChange}
                id={id}
                type={type}
                name={name}
                value={value}
                placeholder={placeholder}
                size={hasSize}
                minLength={hasMin}
                maxLength={hasMax}
                required={required}
                disabled={disabled}
                style={style}
                className={`input-component ${styles.input} ${disabledInput} ${className}`}
            />
        );
    }

    private onChange = (e: React.ChangeEvent<HTMLInputElement>) => {
        !this.props.disabled && this.props.onChange && this.props.onChange(e);
    }

}

export default Input;