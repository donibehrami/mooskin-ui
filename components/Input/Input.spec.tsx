import * as React from 'react';
import Input from './Input';

import { mount, shallow } from 'enzyme';

describe('Input', () => {

    test('renders correctly', () => {
        const func = jest.fn();
        Date.now = jest.fn(() => 1482363367071);
        Math.random = jest.fn(() => 222333444555);

        const tree = shallow(
            <Input
                onChange={func}
                disabled
                className="myClass"
                style={{color: 'blue'}}
                id={'input1'}
                value="input value"
                placeholder="input placeholder"
                maxlength={50}
                minlength={10}
                name="input name"
                type="text"
                description="just some input"
                label="with a label ofc"
                labelWidth={30}
                autofocus
                autocomplete
                required
            />
        );
        expect(tree).toMatchSnapshot();
    });

    test('renders properly into dom and has Placeholder prop', () => {

        const component = shallow(<Input placeholder="username"/>);

        expect(component.find('input').prop('placeholder')).toContain('username');
    });

    test('renders an input with a "required" prop and minlength', () => {
        const component = shallow(<Input minlength={5} required/>);

        expect(component.find('input').prop('required')).toEqual(true);
        expect(component.find('input').prop('minLength')).toEqual(5);
    });

    test('renders an input with a passed value and maxlength', () => {
        const component = shallow(<Input value="random" maxlength={50}/>);

        expect(component.find('input').prop('value')).toEqual('random');
        expect(component.find('input').prop('maxLength')).toEqual(50);
    });

    test('renders an input with id and type', () => {

        const component = shallow(<Input type="text" id="1234"/>);

        expect(component.find('input').prop('id')).toEqual('1234');
        expect(component.find('input').prop('type')).toEqual('text');
    });

    test('renders an input with custom css class and style', () => {
        const component = shallow(<Input style={{color: 'blue'}} className="input-group"/>);

        expect(component.find('input').hasClass('input-group')).toBe(true);
        expect(component.find('input').prop('style')).toEqual({color: 'blue'});
    });

    test('onChange prop callback is called when a key is pressed', () => {
        const func = jest.fn();

        const component = shallow(<Input onChange={func}/>);

        component.find('input').simulate('change', { target: { value: 'text' }});
        expect(func).toHaveBeenCalled();
    });

    test('error classes get applied when input is not validated', () => {
        let inputValue = '';
        let status = null;

        const onChange = (e, data) => {
            inputValue = data.value;
        };

        const validate = (data) => {
            if (data.value < 5){
                status = 'error';
                return false;
            }
            return true;
        };

        const component = mount(<Input status={status} validate={validate} onChange={onChange} value={inputValue}/>);

        component.find('input').simulate('blur');
        expect(component.find('.error')).toBeTruthy();

        component.find('input').simulate('change', { target: { value: 'text' }});
        component.setProps({status: 'error'});
        expect(component.find('.error')).toBeTruthy();

        component.find('input').simulate('change', { target: { value: 'more text' }});
        expect(component.find('.error')).toBeFalsy;
    });

    test('Icon is valid when icon prop is passed', () => {

        const component = shallow(<Input icon="check" />);

        expect(component.find('.icon').length).toBe(1);
    });

});
