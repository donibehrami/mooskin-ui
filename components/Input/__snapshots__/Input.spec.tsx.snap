// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input renders correctly 1`] = `
<div
  className="input-component myClass"
>
  <label
    className="inputLabel"
  >
    with a label ofc
    <input
      className="input input disabledInput"
      disabled={true}
      id="input1"
      maxLength={50}
      minLength={10}
      name="input name"
      onChange={[Function]}
      placeholder="input placeholder"
      required={true}
      style={
        Object {
          "color": "blue",
          "marginLeft": "30px",
        }
      }
      type="text"
      value="input value"
    />
    <i>
      just some input
    </i>
  </label>
</div>
`;
